dbms/src/Storages/Page/V3/LogFile/LogWriter.cpp

```diff
@@ -92,40 +99,32 @@ void LogWriter::close()
 void LogWriter::addRecord(ReadBuffer & payload, const size_t payload_size, const WriteLimiterPtr & write_limiter)
 {
     // Header size varies depending on whether we are recycling or not.
-    const int header_size = recycle_log_files ? Format::RECYCLABLE_HEADER_SIZE : Format::HEADER_SIZE;
+    const UInt32 header_size = recycle_log_files ? Format::RECYCLABLE_HEADER_SIZE : Format::HEADER_SIZE;
 
     // Fragment the record if necessary and emit it. Note that if payload is empty,
     // we still want to iterate once to emit a single zero-length record.
     bool begin = true;
     size_t payload_left = payload_size;
-
-    size_t head_sizes = ((payload_size / Format::BLOCK_SIZE) + 1) * Format::RECYCLABLE_HEADER_SIZE;
-    if (payload_size + head_sizes >= buffer_size)
+    // Padding current block if needed
+    block_offset = block_offset % Format::BLOCK_SIZE; // 0 <= block_offset < Format::BLOCK_SIZE
+    size_t leftover = Format::BLOCK_SIZE - block_offset;
+    assert(leftover > 0);
```

Comment:
Better to use `RUNTIME_ASSERT`

Reply:
This should be just a reminder for later maintenance. Because the condition`Format::BLOCK_SIZE - block_offset % Format::BLOCK_SIZE > 0` is really a simple mathematics and must be true.

Reply:
It is more like a code comment and stronger than code comment. And `RUNTIME_ASSERT` is a little heavy and unnecessary.